packages/launch-config/src/launch-config-crud/create.ts

```diff
@@ -99,79 +120,85 @@ export function updateWorkspaceFoldersIfNeeded(
-        if (fs.existsSync(launchJSONPath) && !appNotInWorkspace) {
-            const existingLaunchConfig = parse(fs.readFileSync(launchJSONPath, 'utf-8')) as LaunchJSON;
-            const updatedConfigurations = existingLaunchConfig.configurations.concat(
-                launchJsonFile?.configurations ?? []
-            );
-            fs.writeFileSync(
-                launchJSONPath,
-                JSON.stringify({ ...existingLaunchConfig, configurations: updatedConfigurations }, null, 4)
-            );
-        } else {
-            const dotVscodePath = join(rootFolderPath, DirName.VSCode);
-            fs.mkdirSync(dotVscodePath, { recursive: true });
-            const path = join(dotVscodePath, 'launch.json');
-            fs.writeFileSync(path, JSON.stringify(launchJsonFile ?? {}, null, 4), 'utf8');
-        }
-    } catch (error) {
-        log?.error(t('errorLaunchFile', { error: error }));
+): Promise<Editor> {
+    const { launchJsonPath, workspaceFolderUri, cwd, appNotInWorkspace } = handleWorkspaceConfig(
+        rootFolder,
+        debugOptions
+    );
+    const configurations = configureLaunchJsonFile(rootFolder, cwd, debugOptions).configurations;
+
+    const npmCommand = debugOptions.datasourceType === ProjectDataSourceType.metadataFile ? 'run start-mock' : 'start';
+    log?.info(
+        t('startServerMessage', {
+            folder: basename(rootFolder),
+            npmCommand
+        })
+    );
+    const launchJsonWritePath = join(launchJsonPath, DirName.VSCode, LAUNCH_JSON_FILE);
+    if (fs.exists(launchJsonWritePath)) {
+        await handleExistingLaunchJson(fs, launchJsonWritePath, configurations, appNotInWorkspace);
+    } else {
+        writeLaunchJsonFile(fs, launchJsonWritePath, configurations);
     }
-    updateWorkspaceFoldersIfNeeded(updateWorkspaceFolders, rootFolderPath, log);
+
+    // The `workspaceFolderUri` is a URI obtained from VS Code that specifies the path to the workspace folder.
+    // This URI is populated when a reload of the workspace is required. It allows us to identify and update
+    // the workspace folder correctly within VS Code.
+    const updateWorkspaceFolders = workspaceFolderUri
+        ? ({
+              uri: workspaceFolderUri,
+              projectName: basename(rootFolder),
+              vscode: debugOptions.vscode
+          } as UpdateWorkspaceFolderOptions)
+        : undefined;
+
+    updateWorkspaceFoldersIfNeeded(updateWorkspaceFolders);
+    return fs;
 }
 
 /**
- * Generates and creates launch configuration for the project based on debug options.
+ * Enhance or create the launch.json file with new launch config.
  *
- * @param {DebugOptions} options - The options for configuring the debug setup.
- * @param log - The logger instance.
+ * @param rootFolder - workspace root folder.
+ * @param fioriOptions - options for the new launch config.
+ * @param fs - optional, the memfs editor instance.
+ * @param log
```

Comment:
description for parameter `log` is missing. By the way, perhaps `logger` would be more intuitive.

Reply:
changed to `logger`